controller_manager:
  ros__parameters:
    update_rate: 50  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    forward_position_controller:
      type: forward_command_controller/ForwardCommandController
 
    joint_trajectory_position_controller:
      type: joint_trajectory_controller/JointTrajectoryController
 
    forward_hand_controller:  
      type: forward_command_controller/ForwardCommandController
 
    joint_trajectory_hand_controller:
      type: joint_trajectory_controller/JointTrajectoryController
 
    fts_broadcaster:
      type: force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster


########################################
# arm position controllers

forward_position_controller:
  ros__parameters:
    joints:
      - shoulder_joint
      - upper_arm_joint
      - middle_arm_joint
      - fore_arm_joint
    interface_name: position

joint_trajectory_position_controller:
  ros__parameters:
    joints:
      - shoulder_joint
      - upper_arm_joint
      - middle_arm_joint
      - fore_arm_joint

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0                            # Defaults to 50
    action_monitor_rate: 20.0                           # Defaults to 20

    allow_partial_joints_goal: false                    # Defaults to false
    open_loop_control: false
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01                  # Defaults to 0.01
      goal_time: 0.0                                    # Defaults to 0.0 (start immediately)


########################################
# hand controllers

forward_hand_controller:
  ros__parameters:
    joints:
      - effector_joint
      - finger_left_joint
      - finger_right_joint
    interface_name: position    

joint_trajectory_hand_controller:
  ros__parameters:
    joints:
      - effector_joint
      - finger_left_joint
      - finger_right_joint

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0                            # Defaults to 50
    action_monitor_rate: 20.0                           # Defaults to 20

    allow_partial_joints_goal: false                    # Defaults to false
    open_loop_control: false
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01                  # Defaults to 0.01
      goal_time: 0.0                                    # Defaults to 0.0 (start immediately)


########################################
# sensors

fts_broadcaster:
  ros__parameters:
    interface_names.force.x: tcp_fts_sensor/force.x
    interface_names.torque.z: tcp_fts_sensor/torque.z
    frame_id: tool_link